File: src/main/java/mc/rpgstats/command/SkillSuggestionProvider.java
package mc.rpgstats.command;

import com.mojang.brigadier.context.CommandContext;
import com.mojang.brigadier.exceptions.CommandSyntaxException;
import com.mojang.brigadier.suggestion.SuggestionProvider;
import com.mojang.brigadier.suggestion.Suggestions;
import com.mojang.brigadier.suggestion.SuggestionsBuilder;
import mc.rpgstats.main.CustomComponents;
import net.minecraft.server.command.ServerCommandSource;
import net.minecraft.util.Identifier;

import java.util.concurrent.CompletableFuture;

public class SkillSuggestionProvider implements SuggestionProvider<ServerCommandSource> {
    @Override
    public CompletableFuture<Suggestions> getSuggestions(CommandContext<ServerCommandSource> context, SuggestionsBuilder builder) throws CommandSyntaxException {
        for (Identifier skillID : CustomComponents.oldComponentIdToComponentIndexMap.keySet()) {
            builder.suggest('"' + skillID.toString() + '"');
        }
        return builder.buildFuture();
    }
}

File: src/main/kotlin/io/github/silverandro/rpgstats/commands/SkillSuggestionProvider.kt
package io.github.silverandro.rpgstats.commands

import com.mojang.brigadier.context.CommandContext
import com.mojang.brigadier.suggestion.SuggestionProvider
import com.mojang.brigadier.suggestion.Suggestions
import com.mojang.brigadier.suggestion.SuggestionsBuilder
import io.github.silverandro.rpgstats.stats.Components
import net.minecraft.server.command.ServerCommandSource
import java.util.concurrent.CompletableFuture

class SkillSuggestionProvider : SuggestionProvider<ServerCommandSource> {
    override fun getSuggestions(
        context: CommandContext<ServerCommandSource>,
        builder: SuggestionsBuilder
    ): CompletableFuture<Suggestions> {
        for (skillID in Components.components.keys) {
            builder.suggest(skillID.toString())
        }
        return builder.buildFuture()
    }
}